libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -74,6 +74,10 @@
 #define HAL_BARO_ALLOW_INIT_NO_BARO
 #endif
 
+#ifndef AP_FIELD_ELEVATION_ENABLED
+#define AP_FIELD_ELEVATION_ENABLED !defined(HAL_BUILD_AP_PERIPH) && !APM_BUILD_TYPE(APM_BUILD_ArduSub)
```

Comment:
Turning the thing off for AP_Periph should be happening in `libraries/AP_HAL_ChibiOS/hwdef/scripts/defaults_periph.h`